File: src/ClientModel/EnumType.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using AutoRest.Core.Model;

namespace AutoRest.Java.Model
{
    /// <summary>
    /// The details of an enumerated type that is used by a service.
    /// </summary>
    public class EnumType : IType
    {
        /// <summary>
        /// Create a new Enum with the provided properties.
        /// </summary>
        /// <param name="name">The name of the new Enum.</param>
        /// <param name="subpackage">The subpackage that this Enum will appear in.</param>
        /// <param name="expandable">Whether or not this will be an ExpandableStringEnum type.</param>
        /// <param name="values">The values of the Enum.</param>
        public EnumType(string package, string name, bool expandable, IEnumerable<ServiceEnumValue> values)
        {
            Name = name;
            Package = package;
            Expandable = expandable;
            Values = values;
        }

        /// <summary>
        /// The name of the new Enum.
        /// </summary>
        public string Name { get; }

        /// <summary>
        /// The package that this enumeration belongs to.
        /// </summary>
        public string Package { get; }

        /// <summary>
        /// Whether or not this will be an ExpandableStringEnum type.
        /// </summary>
        public bool Expandable { get; }

        /// <summary>
        /// The values of the Enum.
        /// </summary>
        public IEnumerable<ServiceEnumValue> Values { get; }

        public void AddImportsTo(ISet<string> imports, bool includeImplementationImports)
        {
            imports.Add($"{Package}.{Name}");
        }

        public IType AsNullable()
        {
            return this;
        }

        public bool Contains(IType type)
        {
            return this == type;
        }

        public string DefaultValueExpression(string sourceExpression)
        {
            return sourceExpression;
        }

        public IType ConvertToClientType()
        {
            return this;
        }

        public override string ToString()
        {
            return Name;
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/clientmodel/EnumType.java
package com.azure.autorest.model.clientmodel;

import java.util.List;
import java.util.Set;

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.


/** 
 The details of an enumerated type that is used by a service.
*/
public class EnumType implements IType
{
    /**
     Create a new Enum with the provided properties.

     @param name The name of the new Enum.
     @param expandable Whether or not this will be an ExpandableStringEnum type.
     @param values The values of the Enum.
    */
    public EnumType(String package_Keyword, String name, boolean expandable, List<ClientEnumValue> values)
    {
        Name = name;
        Package = package_Keyword;
        Expandable = expandable;
        Values = values;
    }

    /**
     The name of the new Enum.
    */
    private String Name;
    public final String getName()
    {
        return Name;
    }

    /**
     The package that this enumeration belongs to.
    */
    private String Package;
    public final String getPackage()
    {
        return Package;
    }

    /**
     Whether or not this will be an ExpandableStringEnum type.
    */
    private boolean Expandable;
    public final boolean getExpandable()
    {
        return Expandable;
    }

    /**
     The values of the Enum.
    */
    private List<ClientEnumValue> Values;
    public final List<ClientEnumValue> getValues()
    {
        return Values;
    }

    public final void AddImportsTo(Set<String> imports, boolean includeImplementationImports)
    {
        imports.add(String.format("%1$s.%2$s", getPackage(), getName()));
    }

    public final IType AsNullable()
    {
        return this;
    }

    public final boolean Contains(IType type)
    {
        return this == type;
    }

    public final String DefaultValueExpression(String sourceExpression)
    {
        if (sourceExpression == null)
        {
            return null;
        }
        return String.format("%1$s.%2$s", getName(), sourceExpression.toUpperCase());
    }

    public final IType getClientType()
    {
        return this;
    }

    public final String ConvertToClientType(String expression)
    {
        return expression;
    }

    public final String ConvertFromClientType(String expression)
    {
        return expression;
    }

    @Override
    public String toString()
    {
        return getName();
    }
}